### PR TITLE
Containing all gci-2014 commits

### PR DESCRIPTION
**Pull request by k-joseph**
This PR contains all commits previously merged into https://github.com/k-joseph/openmrs-core/tree/gci-2014

### CODE REVIEW

File: api/src/main/java/org/openmrs/Cohort.java [hunk 1]
```diff
@@ -27,13 +27,14 @@
 import org.simpleframework.xml.Element;
 import org.simpleframework.xml.ElementList;
 import org.simpleframework.xml.Root;
+import org.springframework.expression.EvaluationContext;
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/Cohort.java]: Seems this is not used by the introduce changes!

File: api/src/main/java/org/openmrs/Concept.java [hunk 2]
```diff
@@ -973,11 +974,10 @@ public void setShortName(ConceptName shortName) {
 				oldShortName.setConceptNameType(null);
 			}
 			shortName.setConceptNameType(ConceptNameType.SHORT);
-			if (StringUtils.isNotBlank(shortName.getName())) {
+			if (StringUtils.isNotBlank(shortName.getName())
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/Concept.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/Concept.java [hunk 3]
```diff
@@ -973,11 +974,10 @@ public void setShortName(ConceptName shortName) {
 				oldShortName.setConceptNameType(null);
 			}
 			shortName.setConceptNameType(ConceptNameType.SHORT);
-			if (StringUtils.isNotBlank(shortName.getName())) {
+			if (StringUtils.isNotBlank(shortName.getName())
+			        && (shortName.getConceptNameId() == null || !getNames().contains(shortName))) {
 				//add this name, if it is new or not among this concept's names
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/Concept.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/Concept.java [hunk 4]
```diff
@@ -1124,11 +1124,10 @@ public ConceptName getShortestName(Locale locale, Boolean exact) {
...
-				if (possibleName.getLocale().equals(locale)) {
-					if ((shortestNameForLocale == null)
-					        || (possibleName.getName().length() < shortestNameForLocale.getName().length())) {
-						shortestNameForLocale = possibleName;
-					}
+				if (possibleName.getLocale().equals(locale)
+				        && ((shortestNameForLocale == null) || (possibleName.getName().length() < shortestNameForLocale
+				                .getName().length()))) {
+					shortestNameForLocale = possibleName;
 				}
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/Concept.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/ConceptAnswer.java [hunk 5]
```diff
@@ -251,4 +251,13 @@ public int compareTo(ConceptAnswer ca) {
 		}
 		return (getSortWeight() < ca.getSortWeight()) ? -1 : (getSortWeight() > ca.getSortWeight()) ? 1 : 0;
 	}
+
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/ConceptAnswer.java]: Do we need this equals. Seems default equals does the same thing.

File: api/src/main/java/org/openmrs/ConceptName.java [hunk 6]
```diff
@@ -526,10 +526,9 @@ public Boolean hasTag(String tagToFind) {
...
 	 */
 	public Boolean isPreferredInLanguage(String language) {
-		if (!StringUtils.isBlank(language) && this.locale != null) {
-			if (isPreferred() && this.locale.getLanguage().equals(language)) {
-				return true;
-			}
+		if (!StringUtils.isBlank(language) && this.locale != null && isPreferred()
+		        && this.locale.getLanguage().equals(language)) {
+			return true;
 		}
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/ConceptName.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/ConceptName.java [hunk 7]
```diff
@@ -545,10 +544,9 @@ public Boolean isPreferredInLanguage(String language) {
...
 	 */
 	public Boolean isPreferredInCountry(String country) {
-		if (!StringUtils.isBlank(country) && this.locale != null) {
-			if (isPreferred() && this.locale.getCountry().equals(country)) {
-				return true;
-			}
+		if (!StringUtils.isBlank(country) && this.locale != null && isPreferred()
+		        && this.locale.getCountry().equals(country)) {
+			return true;
 		}
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/ConceptName.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/ConceptName.java [hunk 8]
```diff
@@ -647,10 +645,8 @@ public Boolean isSynonym() {
 	 */
 	@Deprecated
 	public Boolean isPreferredShortInLanguage(String language) {
-		if (!StringUtils.isBlank(language) && this.locale != null) {
-			if (isShort() && this.locale.getLanguage().equals(language)) {
-				return true;
-			}
+		if (!StringUtils.isBlank(language) && this.locale != null && isShort() && this.locale.getLanguage().equals(language)) {
+			return true;
 		}
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/ConceptName.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/ConceptName.java [hunk 9]
```diff
@@ -667,10 +663,8 @@ public Boolean isPreferredShortInLanguage(String language) {
 	 */
 	@Deprecated
 	public Boolean isPreferredShortInCountry(String country) {
-		if (!StringUtils.isBlank(country) && this.locale != null) {
-			if (isShort() && this.locale.getCountry().equals(country)) {
-				return true;
-			}
+		if (!StringUtils.isBlank(country) && this.locale != null && isShort() && this.locale.getCountry().equals(country)) {
+			return true;
 		}
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/ConceptName.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/ConceptReferenceTerm.java [hunk 10]
```diff
@@ -169,16 +169,15 @@ public void setId(Integer id) {
 	 * @should not add duplicate concept reference term maps
 	 */
 	public void addConceptReferenceTermMap(ConceptReferenceTermMap conceptReferenceTermMap) {
-		if (conceptReferenceTermMap != null) {
+		if (conceptReferenceTermMap != null && conceptReferenceTermMap.getTermB() != null
+		        && !this.equals(conceptReferenceTermMap.getTermB())) {
 			//can't map a term to itself
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/ConceptReferenceTerm.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/ConceptSet.java [hunk 11]
```diff
@@ -206,4 +206,13 @@ public int compareTo(ConceptSet cs) {
 		}
 		return value;
 	}
+
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/ConceptSet.java]: Default equlas does the same right?

File: api/src/main/java/org/openmrs/FormField.java [hunk 12]
```diff
@@ -136,6 +136,15 @@ public int compareTo(FormField f) {
 		
 	}
 	
+	@Override
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/FormField.java]: Same here!

File: api/src/main/java/org/openmrs/PatientIdentifier.java [hunk 13]
```diff
@@ -238,6 +238,15 @@ public int compareTo(PatientIdentifier other) {
 		return retValue;
 	}
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/PatientIdentifier.java]: Same here for equlas

File: api/src/main/java/org/openmrs/PatientState.java [hunk 14]
```diff
@@ -206,4 +206,13 @@ public int compareTo(PatientState o) {
 		}
 		return result;
 	}
+	
+	@Override
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/PatientState.java]: same here for equals

File: api/src/main/java/org/openmrs/PersonAttribute.java [hunk 15]
```diff
@@ -322,6 +322,15 @@ public int compareTo(PersonAttribute other) {
 		return OpenmrsUtil.compareWithNullAsGreatest(getPersonAttributeId(), other.getPersonAttributeId());
 	}
 	
+	@Override
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/PersonAttribute.java]: Same for equals

File: api/src/main/java/org/openmrs/PersonAttributeType.java [hunk 16]
```diff
@@ -183,4 +183,13 @@ public void setId(Integer id) {
 	public int compareTo(PersonAttributeType other) {
 		return OpenmrsUtil.compareWithNullAsGreatest(getPersonAttributeTypeId(), other.getPersonAttributeTypeId());
 	}
+
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/PersonAttributeType.java]: Same

File: api/src/main/java/org/openmrs/activelist/Problem.java [hunk 17]
```diff
@@ -120,4 +120,13 @@ public int compareTo(Problem item) {
 		
 		return mySW.compareTo(theirSW);
 	}
+
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/activelist/Problem.java]: Same here on equals

File: api/src/main/java/org/openmrs/annotation/OpenmrsProfileExcludeFilter.java [hunk 18]
```diff
@@ -54,10 +54,9 @@ public boolean match(MetadataReader metadataReader, MetadataReaderFactory metada
...
 	public boolean matchOpenmrsProfileAttributes(Map<String, Object> openmrsProfile) {
 		Object openmrsVersion = openmrsProfile.get("openmrsVersion");
-		if (StringUtils.isNotBlank((String) openmrsVersion)) {
-			if (!ModuleUtil.matchRequiredVersions(OpenmrsConstants.OPENMRS_VERSION_SHORT, (String) openmrsVersion)) {
-				return false;
-			}
+		if (StringUtils.isNotBlank((String) openmrsVersion)
+		        && !ModuleUtil.matchRequiredVersions(OpenmrsConstants.OPENMRS_VERSION_SHORT, (String) openmrsVersion)) {
+			return false;
 		}
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/annotation/OpenmrsProfileExcludeFilter.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/annotation/OpenmrsProfileExcludeFilter.java [hunk 19]
```diff
@@ -69,11 +68,10 @@ public boolean matchOpenmrsProfileAttributes(Map<String, Object> openmrsProfile)
...
-				if (module.getModuleId().equals(moduleId)) {
-					if (ModuleUtil.matchRequiredVersions(module.getVersion(), moduleVersion)) {
-						moduleMatched = true;
-						break;
-					}
+				if (module.getModuleId().equals(moduleId)
+				        && ModuleUtil.matchRequiredVersions(module.getVersion(), moduleVersion)) {
+					moduleMatched = true;
+					break;
 				}
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/annotation/OpenmrsProfileExcludeFilter.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/api/context/UserContext.java [hunk 20]
```diff
@@ -309,13 +309,12 @@ public Locale getLocale() {
 	public boolean hasPrivilege(String privilege) {
 		
 		// if a user has logged in, check their privileges
-		if (isAuthenticated()) {
+		if (isAuthenticated()
+		        && (getAuthenticatedUser().hasPrivilege(privilege) || getAuthenticatedRole().hasPrivilege(privilege))) {
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/api/context/UserContext.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/api/db/hibernate/AuditableInterceptor.java [hunk 21]
```diff
@@ -165,13 +165,9 @@ private boolean changePropertyValue(Object[] currentState, String[] propertyName
...
-			if (currentState[index] == null || !setNullOnly) {
-				if (!value.equals(currentState[index])) {
-					currentState[index] = value;
-					return true;
-				}
-			}
+		if (index >= 0 && (currentState[index] == null || !setNullOnly) && !value.equals(currentState[index])) {
+			currentState[index] = value;
+			return true;
 		}
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/api/db/hibernate/AuditableInterceptor.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/api/handler/BaseUnretireHandler.java [hunk 22]
```diff
@@ -54,14 +54,13 @@
 	public void handle(Retireable retireableObject, User retiringUser, Date origParentRetiredDate, String unused) {
 		
 		// only act on retired objects
-		if (retireableObject.isRetired()) {
+		if (retireableObject.isRetired()
+		        && (origParentRetiredDate == null || origParentRetiredDate.equals(retireableObject.getDateRetired()))) {
 			// only act on retired objects that match the same date retired as the parent
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/api/handler/BaseUnretireHandler.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/api/handler/BaseUnvoidHandler.java [hunk 23]
```diff
@@ -54,15 +54,14 @@
 	public void handle(Voidable voidableObject, User voidingUser, Date origParentVoidedDate, String unused) {
 		
 		// only operate on voided objects
-		if (voidableObject.isVoided()) {
+		if (voidableObject.isVoided()
+		        && (origParentVoidedDate == null || origParentVoidedDate.equals(voidableObject.getDateVoided()))) {
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/api/handler/BaseUnvoidHandler.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/api/handler/OpenmrsObjectSaveHandler.java [hunk 24]
```diff
@@ -111,11 +111,9 @@ public void handle(OpenmrsObject openmrsObject, User creator, Date dateCreated,
 					continue;
 				}
 				
-				if ("".equals(value)) {
+				if ("".equals(value) && !(openmrsObject instanceof Voidable && ((Voidable) openmrsObject).isVoided())) {
 					//Set to null only if object is not already voided
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/api/handler/OpenmrsObjectSaveHandler.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/api/impl/DatatypeServiceImpl.java [hunk 25]
```diff
@@ -60,7 +60,7 @@ public void setDao(DatatypeDAO dao) {
 	 */
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/api/impl/DatatypeServiceImpl.java]: Do we need really need these sysnchornized blocks?

File: api/src/main/java/org/openmrs/attribute/BaseAttribute.java [hunk 26]
```diff
@@ -149,4 +149,12 @@ public int compareTo(Attribute other) {
 		return retValue;
 	}
 	
+	@Override
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/attribute/BaseAttribute.java]: Do we need this equals?

File: api/src/main/java/org/openmrs/layout/LayoutSupport.java [hunk 27]
```diff
@@ -59,13 +59,12 @@ public T getLayoutTemplateByName(String templateName) {
 			T ret = null;
 			
 			for (T at : this.layoutTemplates) {
-				if (at != null) {
-					if (templateName.equalsIgnoreCase(at.getDisplayName())
-					        || templateName.equalsIgnoreCase(at.getCodeName())
-					        || templateName.equalsIgnoreCase(at.getCountry())) {
-						ret = at;
-						log.debug("Found Layout Template named " + at.getDisplayName());
-					}
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/layout/LayoutSupport.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/layout/LayoutSupport.java [hunk 28]
```diff
@@ -81,11 +80,9 @@ public T getLayoutTemplateByCodeName(String templateName) {
 			T ret = null;
 			
 			for (T at : this.layoutTemplates) {
-				if (at != null) {
-					if (templateName.equalsIgnoreCase(at.getCodeName())) {
-						ret = at;
-						log.debug("Found Layout Template named " + at.getDisplayName());
-					}
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/layout/LayoutSupport.java]: Code make complex by aggregating this conditions. Should we have this or not?

File: api/src/main/java/org/openmrs/obs/handler/TextHandler.java [hunk 29]
```diff
@@ -132,6 +132,7 @@ public Obs saveObs(Obs obs) throws APIException {
 						}
 						fout.write(character);
 					}
+					tempRd.close();
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/obs/handler/TextHandler.java]: better do it in a finally block

File: api/src/main/java/org/openmrs/util/UpgradeUtil.java [hunk 30]
```diff
@@ -43,15 +43,18 @@ public static Integer getConceptIdForUnits(String units) {
...
-			        + DatabaseUtil.ORDER_ENTRY_UPGRADE_SETTINGS_FILENAME));
+			FileInputStream fis = new FileInputStream(appDataDir + System.getProperty("file.separator")
+			        + DatabaseUtil.ORDER_ENTRY_UPGRADE_SETTINGS_FILENAME);
+			props.load(fis);
 			for (Map.Entry prop : props.entrySet()) {
 				if (prop.getKey().equals(units)) {
 					conceptId = prop.getValue().toString();
 					
 					if (conceptId != null) {
+						fis.close();
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/util/UpgradeUtil.java]: Need to close streams in a finally blocks

File: api/src/main/java/org/openmrs/util/UpgradeUtil.java [hunk 31]
```diff
@@ -43,15 +43,18 @@ public static Integer getConceptIdForUnits(String units) {
...
 						return Integer.valueOf(conceptId);
 					} else {
+						fis.close();
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/util/UpgradeUtil.java]: Need to close streams in a finally blocks

File: web/src/main/java/org/openmrs/web/controller/encounter/EncounterDisplayController.java [hunk 32]
```diff
@@ -390,6 +390,14 @@ public int compareTo(FieldHolder other) {
 			return temp;
 		}
```
Comment (harsha89) [inline web/src/main/java/org/openmrs/web/controller/encounter/EncounterDisplayController.java]: Equals needed?

File: web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java [hunk 33]
```diff
@@ -949,7 +949,9 @@ private boolean verifyConnection(String connectionUsername, String connectionPas
 			// verify connection
 			//Set Database Driver using driver String
 			Class.forName(loadedDriverString).newInstance();
-			DriverManager.getConnection(databaseConnectionFinalUrl, connectionUsername, connectionPassword);
+			Connection tempConnection = DriverManager.getConnection(databaseConnectionFinalUrl, connectionUsername,
```
Comment (harsha89) [inline web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java]: Need to close conncetion in finaly block

File: web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java [hunk 34]
```diff
@@ -1132,7 +1134,10 @@ private int executeStatement(boolean silent, String user, String pw, String sql,
 			
 			// run the sql statement
 			Statement statement = connection.createStatement();
-			return statement.executeUpdate(replacedSql);
+			
+			int updateDelta = statement.executeUpdate(replacedSql);
+			statement.close();
```
Comment (harsha89) [inline web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java]: close connections in a finally block

File: web/src/main/java/org/openmrs/web/filter/update/UpdateFilter.java [hunk 35]
```diff
@@ -350,9 +350,13 @@ protected boolean authenticateAsSuperUser(String usernameOrSystemId, String pass
...
 					String salt = results.getString(3);
 					String passwordToHash = password + salt;
-					return Security.hashMatches(storedPassword, passwordToHash) && isSuperUser(connection, userId);
+					boolean result = Security.hashMatches(storedPassword, passwordToHash) && isSuperUser(connection, userId);
+					statement.close();
+					return result;
 				}
 			}
+			// Close statement
+			statement.close();
```
Comment (harsha89) [inline web/src/main/java/org/openmrs/web/filter/update/UpdateFilter.java]: Stmt should close in finally blocls

File: web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java [hunk 36]
```diff
@@ -71,6 +71,9 @@ public static String restoreLocale(String username) {
 							currentLocale = results.getString(1);
 						}
 					}
+					
+					//close statement
+					statement.close();
```
Comment (harsha89) [inline web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java]: same as above

File: web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java [hunk 37]
```diff
@@ -129,14 +132,12 @@ public static String readSystemDefaultLocale(Connection connection) {
...
-				if (connection != null) {
-					try {
-						connection.close();
-					}
-					catch (SQLException e) {
-						log.debug("Error while closing the database", e);
-					}
+			if (needToCloseConection && connection != null) {
+				try {
+					connection.close();
```
Comment (harsha89) [inline web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java]: same as abve

File: web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java [hunk 38]
```diff
@@ -180,6 +181,8 @@ public static boolean storeLocale(String locale) {
 				if (statement.executeUpdate() != 1) {
 					log.warn("Unable to set system default locale property.");
 				}
+				// close statement to release resources
+				statement.close();
```
Comment (harsha89) [inline web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java]: same as abve

File: api/src/main/java/org/openmrs/Cohort.java (hunk 1 above)
Comment (k-joseph) [inline api/src/main/java/org/openmrs/Cohort.java]: Actually shows up when i use the Ctrl + Shift + O option for resolving import issues

File: api/src/main/java/org/openmrs/Concept.java (hunk 3 above)
Comment (k-joseph) [inline api/src/main/java/org/openmrs/Concept.java]: This is not gci, it is https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/Concept.java#L978 and am ok with it

File: api/src/main/java/org/openmrs/Concept.java (hunk 4 above)
Comment (k-joseph) [inline api/src/main/java/org/openmrs/Concept.java]: This looks to be just the same thing and my above response is ok, remembering that i created this Pull Request after checking out a new branch from master. Then pulled from upstream/gci-2014 and resolved conflicts which was the best way to confirm that every thing is updated and eliminate more conflicts

File: api/src/main/java/org/openmrs/ConceptAnswer.java (hunk 5 above)
Comment (k-joseph) [inline api/src/main/java/org/openmrs/ConceptAnswer.java]: Actually the modification just shows a space which is well and introduced by the openmrs formatter

File: api/src/main/java/org/openmrs/ConceptName.java (hunk 6 above)
Comment (k-joseph) [inline api/src/main/java/org/openmrs/ConceptName.java]: See  my above responses :+1:

File: api/src/main/java/org/openmrs/ConceptName.java (hunk 7 above)
Comment (k-joseph) [inline api/src/main/java/org/openmrs/ConceptName.java]: See  my above responses :+1:

File: api/src/main/java/org/openmrs/ConceptName.java (hunk 8 above)
Comment (k-joseph) [inline api/src/main/java/org/openmrs/ConceptName.java]: See  my above responses :+1:

File: api/src/main/java/org/openmrs/ConceptName.java (hunk 9 above)
Comment (k-joseph) [inline api/src/main/java/org/openmrs/ConceptName.java]: See  my above responses :+1:

File: api/src/main/java/org/openmrs/ConceptReferenceTerm.java (hunk 10 above)
Comment (k-joseph) [inline api/src/main/java/org/openmrs/ConceptReferenceTerm.java]: See  my above responses :+1:

File: api/src/main/java/org/openmrs/ConceptSet.java (hunk 11 above)
Comment (k-joseph) [inline api/src/main/java/org/openmrs/ConceptSet.java]: See  my above responses :+1:

File: api/src/main/java/org/openmrs/PatientState.java (hunk 14 above)
Comment (k-joseph) [inline api/src/main/java/org/openmrs/PatientState.java]: See  my above responses :+1:

**Comment by k-joseph** [comment]
@dkayiwa, my last commit here at: https://github.com/k-joseph/openmrs-core/commit/91c6be799f090fa3b1fe1a9c325cdb4b68da1905 addresses your reviews, we can later handle the log and deprecated issues

**Pull request merged**